src/VirtoCommerce.CatalogPersonalizationModule.Data/Repositories/IPersonalizationRepository.cs

```diff
@@ -1,15 +1,15 @@
-using System.Linq;
 using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
 using VirtoCommerce.CatalogPersonalizationModule.Data.Model;
 using VirtoCommerce.Platform.Core.Common;
 
 namespace VirtoCommerce.CatalogPersonalizationModule.Data.Repositories
 {
     public interface IPersonalizationRepository : IRepository
     {
-        IQueryable<TaggedItemEntity> TaggedItems { get; }
-        IQueryable<TagEntity> Tags { get; }
-        IQueryable<TaggedItemOutlineEntity> TaggedItemOutlines { get; }
+        DbSet<TaggedItemEntity> TaggedItems { get; }
```

Comment:
Excuse me, what was the reason to change from IQueryable? Every member in other places defined in the same manner (as IQueryable).
I know FromSqlInterpolated is a DbSet member. But why just not to use type conversion before calling FromSqlInterpolated?

Reply:
In general, it is better to expose real type in interface than guessing the underlying type of the interface implementation. Though maybe I do not know the reasons why we should hide DbSet.
It would be a violation of Dependency Inversion principle. If IPersonalizationRepository implementation will be changed and overridden in some extension, such cast may just become invalid.

Reply:
The reason why we should not use DbSet is that we do not want a dependency on EF class, which DbSet is, according to the same DI principle.
So added the method for getting outlines that lays inside outlines array to IPersonalisationRepository.